austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,96 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone/
+	var/beacons
+	var/default_hatmask
+	var/obj/item/t_scanner/adv_mining_scanner/lesser/scanner //there is no code to turn this scanner off.
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	weather_immunities += "ash" // no damage from ash storms
+	stored_gun.overheat_time = 10
+	beacons = 15
```

Comment:
You can probably define this with the object rather than initialize

Reply:
To my understanding, I can put (beacons = 15) in the definition but not the other two?  The gun for sure will not exist until we run initialisation.

Reply:
Just to be sure, I've tried it and the code won't build if I try to move the other 2 lines.
But I have moved the beacons line as requested.

Reply:
yeah, only beacons. overheat time references another object and immunities is an operation, both of which can't run on object definitions during compile time.